scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -458,9 +455,9 @@ def test_zero_rhs(solver):
         platform.machine() not in ['x86_64' 'x86', 'aarch64', 'arm64'],
         reason="fails on at least ppc64le, ppc64 and riscv64, see gh-17839")
     ),
```

Comment:
The parametrization outside of the test should be removed, and all the skips should become skips within the test, e.g.

```python
# no more parametrization here
def test_maxiter_worsening(solver):
    if solver == gmres and platform.machine() == 'aarch64' and and sys.version_info[1] == 9:
        pytest.xfail(reason="gh-13019")
    if solver == lgmres and platform.machine() not in ['x86_64' 'x86', 'aarch64', 'arm64']:
        pytest.xfail(reason="fails on at least ppc64le, ppc64 and riscv64, see gh-17839")
    # etc